compositor_pipeline/src/pipeline/encoder/vulkan_h264.rs

```diff
@@ -0,0 +1,106 @@
+use std::{num::NonZero, ops::Deref, sync::Arc};
+
+use compositor_render::{FrameData, OutputFrameFormat};
+use tracing::{error, info};
+use vk_video::{RateControl, Rational, VideoParameters, WgpuTexturesEncoder};
+
+use crate::prelude::*;
+
+use super::{VideoEncoder, VideoEncoderConfig};
+
+pub struct VulkanH264Encoder {
+    encoder: WgpuTexturesEncoder,
+}
+
+impl VideoEncoder for VulkanH264Encoder {
+    const LABEL: &'static str = "Vulkan H264 encoder";
+
+    type Options = VulkanH264EncoderOptions;
+
+    fn new(
+        ctx: &Arc<PipelineCtx>,
+        options: Self::Options,
+    ) -> Result<(Self, VideoEncoderConfig), EncoderInitError> {
+        let Some(vulkan_ctx) = &ctx.graphics_context.vulkan_ctx else {
+            return Err(EncoderInitError::VulkanContextRequiredForVulkanEncoder);
+        };
+
+        info!("Initializing Vulkan H264 encoder");
+        let width = NonZero::new(u32::max(options.resolution.width as u32, 1)).unwrap();
+        let height = NonZero::new(u32::max(options.resolution.height as u32, 1)).unwrap();
+        let framerate = ctx.output_framerate;
+        let bitrate = options.bitrate.unwrap_or_else(|| {
+            let average_bitrate = (width.get() * height.get()) as f64
```

Comment:
Not sure what number it produces, but consider rounding it to something that looks good